c_src/enacl_nif.c

```diff
@@ -1433,6 +1433,100 @@ ERL_NIF_TERM enif_crypto_aead_chacha20poly1305_decrypt(ErlNifEnv *env, int argc,
 
 }
 
+/*
+ * AEAD XChaCha20 Poly1305 IETF
+ */
+
+static
+ERL_NIF_TERM enif_crypto_aead_xchacha20poly1305_ietf_KEYBYTES(ErlNifEnv *env, int argc, ERL_NIF_TERM const argv[]) {
+  return enif_make_int64(env, crypto_aead_xchacha20poly1305_ietf_KEYBYTES);
+}
+
+static
+ERL_NIF_TERM enif_crypto_aead_xchacha20poly1305_ietf_NPUBBYTES(ErlNifEnv *env, int argc, ERL_NIF_TERM const argv[]) {
+  return enif_make_int64(env, crypto_aead_xchacha20poly1305_ietf_NPUBBYTES);
+}
+
+static
+ERL_NIF_TERM enif_crypto_aead_xchacha20poly1305_ietf_ABYTES(ErlNifEnv *env, int argc, ERL_NIF_TERM const argv[]) {
+  return enif_make_int64(env, crypto_aead_xchacha20poly1305_ietf_ABYTES);
+}
+
+static
+ERL_NIF_TERM enif_crypto_aead_xchacha20poly1305_ietf_MESSAGEBYTES_MAX(ErlNifEnv *env, int argc, ERL_NIF_TERM const argv[]) {
+  return enif_make_int64(env, crypto_aead_xchacha20poly1305_ietf_MESSAGEBYTES_MAX);
+}
+
+static
+ERL_NIF_TERM enif_crypto_aead_xchacha20poly1305_ietf_keygen(ErlNifEnv *env, int argc, ERL_NIF_TERM const argv[]) {
+  ErlNifBinary key;
+  if (0 != argc) return enif_make_badarg(env);
+  
+  enif_alloc_binary(crypto_aead_xchacha20poly1305_ietf_KEYBYTES, &key);
+  crypto_aead_xchacha20poly1305_ietf_keygen(key.data);
+	return enif_make_binary(env, &key);
+}
+
+static
+ERL_NIF_TERM enif_crypto_aead_xchacha20poly1305_ietf_nonce(ErlNifEnv *env, int argc, ERL_NIF_TERM const argv[]) {
```

Comment:
Maybe we should lift this to a `nonce()` function based on the same idea. It seems generalizable.

Reply:
well, the idea is to keep it "idiot proof"... So if make a `nonce()` function it will require that the user pass in the correct int (or helper function which returns N). This should not be asking too much, but .... yeah 😄 

 I am also keen on the idea of keeping the function names at this level in line with libsodium. I think what we can do is make an enacl_simple module where we can make things more "automagic" (i.e., solid defaults) and have prettier func names. 

 But maybe you had something else in mind in regards to `nonce/0`?

Reply:
Hmm, you probably meant `nonce/1` .... `nonce(xchacha20poly1305_ietf)` ? Man, I can't wait for this to become the default in some places in libsodium, these names are ugh. 😄

Reply:
Yes, this is somewhat in the lines I thought of. Rather than having half a gazillion nonce functions, keep one which knows about the sizes. I think it eases the construction quite a lot.

Reply:
Cool and very much agree. Will push something up in a bit.